pkg/controller/garbagecollector/metrics.go

```diff
@@ -65,9 +75,47 @@ func Register() {
 		prometheus.MustRegister(EventProcessingLatency)
 		prometheus.MustRegister(DirtyProcessingLatency)
 		prometheus.MustRegister(OrphanProcessingLatency)
+		prometheus.MustRegister(DeletionToWatchLatency)
 	})
 }
 
 func sinceInMicroseconds(clock clock.Clock, start time.Time) float64 {
 	return float64(clock.Since(start).Nanoseconds() / time.Microsecond.Nanoseconds())
 }
+
+func NewDeletionToWatchTracker(limit int) *uidToDeletionTime {
```

Comment:
1. move to new file
2. rename: NewAPIRequestLatencyTracker
3. Method names: RequestSent/Observed